components/fxa-client/src/lib.rs

```diff
@@ -218,6 +218,43 @@ impl FirefoxAccount {
             None => Err(ErrorKind::NoRefreshToken.into()),
         }
     }
+
+    /// Disconnect from the account and optionaly destroy our device record.
+    ///
+    /// **💾 This method alters the persisted account state.**
+    pub fn disconnect(&mut self) -> Result<()> {
+        // Start over the account state.
+        self.state.commands_data.clear();
+        self.state.device_capabilities.clear();
+        self.state.last_handled_command = None;
+        self.state.scoped_keys.clear();
+        self.access_token_cache.clear();
+        self.flow_store.clear();
+        let refresh_token = match self.state.refresh_token {
+            Some(ref token) => token.token.clone(),
+            None => {
+                log::warn!("No refresh token found to disconnect. Skipping.");
+                return Ok(());
+            }
+        };
+        self.state.refresh_token = None;
+        // TODO: make the method infailible as it will be impossible to delete the
```

Comment:
Do we want this @grigoryk? What do you want to happen if the network fails for example?

Reply:
FWIW I'd be happy with an infallible "best effort" attempt here; the alternative is that we don't let you sign out unless you're connected to the network, which seems wrong to me.